thoth/storages/graph/postgres.py

```diff
@@ -1192,6 +1192,36 @@ def get_python_packages(self) -> Set[str]:
         """Retrieve listing of all Python packages known to graph database instance."""
         return set(item[0] for item in self._session.query(PythonPackageVersionEntity.package_name).distinct().all())
 
+    def get_python_package_versions(
+        self,
+        *,
+        start_offset: int = 0,
+        count: int = 10,
+        os_name: str = None,
+        os_version: str =None,
+        python_version: str = None,
+        distinct: bool=False,
+    ) -> List[Tuple[str, str, str]]:
+        """Retrieve Python package versions in Thoth Database."""
+        query = (
+            self._session.query(PythonPackageVersion)
+            .join(PythonPackageIndex)
+            .with_entities(PythonPackageVersion.package_name, PythonPackageVersion.package_version, PythonPackageIndex.url.isnot(None))
```

Comment:
`isnot(None)` not sure about this, WDYT @fridex? :)

Reply:
I think `PythonPackageVersion` should have always fully qualified packages, so I would say it's not needed - but we need to do the fixes with nullable values on syncs we spoke about.